src/common/translateOption.js

```diff
@@ -0,0 +1,18 @@
+// Copyright (C) 2017-2022 Smart code 203358507
+
+const { t } = require('i18next');
+
+const translateOption = (option, translateKeyPrefix = '') => {
+    const translateKey = `${translateKeyPrefix}${option}`;
+    const translateKeyUppercase = translateKey.toUpperCase();
+    const translateValue = t(translateKey);
+    const translateValueUppercase = t(translateKeyUppercase);
+    if (translateKey !== translateValue) {
+        return translateValue;
+    } else if (translateKeyUppercase !== translateValueUppercase) {
+        return translateValueUppercase;
+    }
+    return option.charAt(0).toUpperCase() + option.slice(1);
+};
```

Comment:
This function is relatively hard to follow. If I understand what it's doing correctly, I believe it can be heavily simplified as follows:
```suggestion
const translateOption = (option, translateKeyPrefix = '') => {
    const translateKey = `${translateKeyPrefix}${option}`;
    const translateValue = t(translateKey, {defaultValue: t(translateKey.toUpperCase(), {defaultValue: null})});
    return translateValue ?? option.charAt(0).toUpperCase() + option.slice(1);
};
```

Reply:
Much better thank you !